src/main/java/frc/robot/subsystems/vision/provider/Vision.java

```diff
@@ -0,0 +1,50 @@
+package frc.robot.subsystems.vision.provider;
+
+import java.util.Optional;
+
+/**
+ * Fuck convention
+ */
+public interface Vision {
+
+    //i wish i had records, and i hate data transfer objects. However, this is a necessary sin so that
+    //other developers who are coding in the frc style can use this object
+    class Coordinates {
```

Comment:
bummer WPILib isn't using java14 so this could be a record.

Reply:
Indeed